modules/core/include/opencv2/core/hal/intrin_neon.hpp

```diff
@@ -1206,6 +1207,16 @@ inline v_float16x4 v_cvt_f16(const v_float32x4& a)
 }
 #endif
 
+//! @name Check SIMD support
+//! @{
+//! @brief Check CPU capability of SIMD operation
+static inline bool hasSIMD128()
```

Comment:
shall we also use predefined macros like `__ARM_NEON__` e.g. like:

<pre>
#ifdef __ARM_NEON__
    return true;
#else
   return checkHardwareSupport(CV_CPU_NEON);
#endif
</pre>